examples/retry_command.ts

```diff
@@ -0,0 +1,32 @@
+import { Command, CommandStep, Pipeline, MetaCommand } from '../src';
+import { YamlSerializer } from '../src/serializers/yaml';
+
+class RetryCommand extends MetaCommand {
+    wrapper_script: string;
+    constructor(public retries: number, public c: Command) {
+        super(new Command(c.toString()));
+        this.retries = retries;
+        this.wrapper_script = '~/wrapperscripts/retry_command';
+    }
+
+    protected serialize(): string {
+        return `${this.wrapper_script} ${this.retries} -- ${this.c.toString()}`;
+    }
+}
```

Comment:
```suggestion
class RetryCommand extends Command {

    constructor(retries: number, c: Command) {
        super(new Command(new Array(retries + 1).fill(c.toString()).join(' || ')));
    }
}
```

I think this is all that is needed to make a retry command work.

If we want to go a bit more clean and really use the `serialize` method (I am not actually sure it is needed, then:

```suggestion
class RetryCommand extends Command {

    constructor(private readonly retries: number, c: Command, timeout: number) {
        super(c.command, c.timeout);
    }

    protected serialize(): string {
        return new Array(this.retries + 1).fill(this.command).join(' || ');
    }
}
```

Reply:
yeah, for the purposes of this example I agree